File: samples/android/app/src/main/kotlin/jp/kamusoft/ksdialogs/samples/android/ModelDialogViewModel.kt
package jp.kamusoft.ksdialogs.samples.android

import jp.kamusoft.ksdialogs.DialogViewModel
import jp.kamusoft.ksdialogs.notifier

/**
 * Model Dialog の ViewModel。
 *
 * 結果型は [Boolean] で、完了操作は `true` を報告する。
 * 表示するメッセージは型指定 show の configure から設定される。
 * 結果は中身から渡される報告口ではなく、この ViewModel 自身が `notifier` から報告する
 * (core/ADR-0018)。
 */
internal class ModelDialogViewModel : DialogViewModel<Boolean> {
    /** ダイアログに表示するメッセージ。表示の直前に configure から設定する。 */
    var message: String = ""

    /** 完了を報告する。 */
    fun complete() {
        notifier?.complete(true)
    }

    /** キャンセルを報告する。 */
    fun cancel() {
        notifier?.cancel()
    }
}

File: samples/ios/KsDialogsSample/ModelDialogViewModel.swift
import KsDialogs

/// Model Dialog の ViewModel。
///
/// 結果型は `Bool` で、完了操作は `true` を報告する。
/// 表示するメッセージは型指定 show の configure から設定されるため、状態は UI スレッドに閉じる。
/// 結果は中身から渡される報告口ではなく、この ViewModel 自身が `notifier` から報告する
/// (core/ADR-0018)。
@MainActor
final class ModelDialogViewModel: DialogViewModel {
    typealias Result = Bool

    /// ダイアログに表示するメッセージ。表示の直前に configure から設定する。
    var message: String = ""

    /// ViewModel factory から作れるように、生成は状態なしで行う。
    nonisolated init() {}

    /// 完了を報告する。
    func complete() {
        notifier?.complete(true)
    }

    /// キャンセルを報告する。
    func cancel() {
        notifier?.cancel()
    }
}
